SteamKit2/SteamKit2/Util/Utils.cs

```diff
@@ -49,115 +49,117 @@ public static EOSType GetOSType()
-                                }
-
-                                goto default;
-
-                            case 6:
-                                switch ( ver.Minor )
-                                {
-                                    case 0:
-                                        return EOSType.WinVista; // Also Server 2008
-
-                                    case 1:
-                                        return EOSType.Windows7; // Also Server 2008 R2
-
-                                    case 2:
-                                        return EOSType.Windows8; // Also Server 2012
-
-                                    // Note: The OSVersion property reports the same version number (6.2.0.0) for both Windows 8 and Windows 8.1.- http://msdn.microsoft.com/en-us/library/system.environment.osversion(v=vs.110).aspx
-                                    // In practice, this will only get hit if the application targets Windows 8.1 in the app manifest.
-                                    // See http://msdn.microsoft.com/en-us/library/windows/desktop/dn481241(v=vs.85).aspx for more info.
-                                    case 3:
-                                        return EOSType.Windows81; // Also Server 2012 R2
-                                }
-
-                                goto default;
+                        case 5:
+                            return ver.Minor switch
+                            {
+                                0 => EOSType.Win2000,
+                                1 => EOSType.WinXP,
+                                // Assume nobody runs Windows XP Professional x64 Edition
+                                // It's an edition of Windows Server 2003 anyway.
+                                2 => EOSType.Win2003,
+                                _ => EOSType.WinUnknown,
+                            };
+
+                        case 6:
+                            return ver.Minor switch
+                            {
+                                0 => EOSType.WinVista, // Also Server 2008
+                                1 => EOSType.Windows7, // Also Server 2008 R2
+                                2 => EOSType.Windows8, // Also Server 2012
+                                // Note: The OSVersion property reports the same version number (6.2.0.0) for both Windows 8 and Windows 8.1.- http://msdn.microsoft.com/en-us/library/system.environment.osversion(v=vs.110).aspx
+                                // In practice, this will only get hit if the application targets Windows 8.1 in the app manifest.
+                                // See http://msdn.microsoft.com/en-us/library/windows/desktop/dn481241(v=vs.85).aspx for more info.
+                                3 => EOSType.Windows81, // Also Server 2012 R2
+                                _ => EOSType.WinUnknown,
+                            };
+
+                        case 10:
+                            if( ver.Build >= 22000 )

```

Comment:
Does `case...when` work in this form of switch statement?

Reply:
Yep.